examples/asset/multi_asset_sync.rs

```diff
@@ -0,0 +1,249 @@
+        2 => asset_server.load_acquire("models/AlienCake/cakeBirthday.glb", guard.clone()),
+        3 => asset_server.load_acquire("models/FlightHelmet/FlightHelmet.gltf", guard.clone()),
+        4 => asset_server.load_acquire("models/torus/torus.gltf", guard.clone()),
+        _ => unreachable!(),
+    })));
+    let future = barrier.wait_async();
+    commands.insert_resource(barrier);
+
+    let loading_state = Arc::new(Mutex::new("Loading..".to_owned()));
+    commands.insert_resource(AsyncLoadingState(loading_state.clone()));
+
+    // await the `AssetBarrierFuture`.
+    AsyncComputeTaskPool::get()
+        .spawn(async move {
+            future.await;
+            "Loading Complete!".clone_into(&mut loading_state.lock().unwrap());
+        })
+        .detach();
+
+    // Display the result of async loading.
+    commands
+        .spawn(NodeBundle {
+            style: Style {
+                width: Val::Percent(100.),
+                height: Val::Percent(100.),
+                justify_content: JustifyContent::End,
+
+                ..default()
+            },
+            ..default()
+        })
+        .with_children(|b| {
+            b.spawn(TextBundle {
+                text: Text {
+                    sections: vec![TextSection {
+                        value: "".to_owned(),
+                        style: TextStyle {
+                            font_size: 64.0,
+                            color: Color::BLACK,
+                            ..Default::default()
+                        },
+                    }],
+                    justify: JustifyText::Right,
+                    ..Default::default()
+                },
+                ..Default::default()
+            });
+        });
+
+    // Camera
+    commands.spawn(Camera3dBundle {
+        transform: Transform::from_xyz(10.0, 10.0, 15.0)
+            .looking_at(Vec3::new(0.0, 0.0, 0.0), Vec3::Y),
+        ..default()
+    });
+
+    // Light
+    commands.spawn(DirectionalLightBundle {
+        transform: Transform::from_rotation(Quat::from_euler(EulerRot::ZYX, 0.0, 1.0, -PI / 4.)),
+        directional_light: DirectionalLight {
+            shadows_enabled: true,
+            ..default()
+        },
+        ..default()
+    });
+
+    // Plane
+    commands.spawn((
+        PbrBundle {
+            mesh: meshes.add(Plane3d::default().mesh().size(50000.0, 50000.0)),
+            material: materials.add(Color::srgb(0.7, 0.2, 0.2)),
+            ..default()
+        },
+        Loading,
+    ));
+}
+
+fn wait_on_load(
+    mut commands: Commands,
+    foxes: Res<OneHundredThings>,
+    barrier: Option<Res<AssetBarrier>>,
+    loading: Query<Entity, With<Loading>>,
+    gltfs: Res<Assets<Gltf>>,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+) {
+    if barrier.map(|b| b.is_ready()) != Some(true) {
+        return;
+    };
+    commands.entity(loading.single()).despawn();
```

Comment:
I think using .single is probably not the right pattern here: it's very fragile when users copy this into their game and start refactoring. I would just iterate over the query.

Reply:
Fixed!